arm/Microsoft.Network/azureFirewalls/deploy.bicep

```diff
@@ -0,0 +1,292 @@
+}
+
+module pid_cuaId './.bicep/nested_cuaId.bicep' = if (!empty(cuaId)) {
+  name: 'pid-${cuaId}'
+  params: {}
+}
+
+resource azureFirewallPip 'Microsoft.Network/publicIPAddresses@2021-02-01' = {
+  name: azureFirewallPipName_var
+  location: location
+  tags: tags
+  sku: {
+    name: 'Standard'
+  }
+  zones: availabilityZones
+  properties: {
+    publicIPAllocationMethod: 'Static'
+    publicIPAddressVersion: 'IPv4'
+    publicIPPrefix: ((!empty(publicIPPrefixId)) ? publicIPPrefix : json('null'))
+  }
+}
+
+resource azureFirewallPip_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${azureFirewallPipName_var}-publicIpDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  dependsOn: [
+    azureFirewallPip
+  ]
+}
+
+resource azureFirewallPip_diagnosticSettings 'Microsoft.Insights/diagnosticsettings@2017-05-01-preview' = if ((!empty(diagnosticStorageAccountId)) || (!empty(workspaceId)) || (!empty(eventHubAuthorizationRuleId)) || (!empty(eventHubName))) {
+  name: '${azureFirewallPipName_var}-diagnosticSettings'
+  properties: {
+    storageAccountId: (empty(diagnosticStorageAccountId) ? json('null') : diagnosticStorageAccountId)
+    workspaceId: (empty(workspaceId) ? json('null') : workspaceId)
+    eventHubAuthorizationRuleId: (empty(eventHubAuthorizationRuleId) ? json('null') : eventHubAuthorizationRuleId)
+    eventHubName: (empty(eventHubName) ? json('null') : eventHubName)
+    metrics: ((empty(diagnosticStorageAccountId) && empty(workspaceId) && empty(eventHubAuthorizationRuleId) && empty(eventHubName)) ? json('null') : diagnosticsMetrics)
+    logs: ((empty(diagnosticStorageAccountId) && empty(workspaceId) && empty(eventHubAuthorizationRuleId) && empty(eventHubName)) ? json('null') : diagnosticsLogsPublicIp)
+  }
+  dependsOn: [
+    azureFirewallPip
+  ]
+}
+
+resource azureFirewall 'Microsoft.Network/azureFirewalls@2021-02-01' = {
+  name: azureFirewallName
+  location: location
+  zones: ((length(availabilityZones) == 0) ? json('null') : availabilityZones)
+  tags: tags
+  properties: {
+    threatIntelMode: 'Deny'
+    ipConfigurations: [
+      {
+        name: 'IpConf'
+        properties: {
+          subnet: {
+            id: azureFirewallSubnetId
+          }
+          publicIPAddress: {
+            id: azureFirewallPipId
+          }
+        }
+      }
+    ]
+    sku: {
+      name: azureSkuName
+      tier: azureSkuTier
+    }
+    additionalProperties: {
+      'Network.DNS.EnableProxy': string(enableDnsProxy)
+    }
+    applicationRuleCollections: applicationRuleCollections
+    natRuleCollections: natRuleCollections
+    networkRuleCollections: networkRuleCollections
+  }
+}
+
+resource azureFirewall_lock 'Microsoft.Authorization/locks@2016-09-01' = if (lockForDeletion) {
+  name: '${azureFirewall.name}-azureFirewallDoNotDelete'
+  properties: {
+    level: 'CanNotDelete'
+  }
+  dependsOn: [
```

Comment:
As per Ahmeds comment can be removed as the `.name` reference injects a dependency